src/EnergyPlus/FanCoilUnits.cc

```diff
@@ -4307,48 +4339,89 @@ namespace FanCoilUnits {
-                        ShowContinueErrorTimeStamp("Iterations=" + TrimSigDigits(MaxIterCycl));
+                    Real64 eHeatCoilCyclingR = PLR;
+                    // electric heating coil
+                    if (QUnitOutMax > QZnReq) {
+                        // heating coil output is larger than required, mudulate the electric heating coil output to meet the load
+                        Node(InletNode).MassFlowRateMinAvail = Node(InletNode).MassFlowRate;
+                        Par(1) = double(FanCoilNum);
+                        Par(2) = 0.0; // FLAG, IF 1.0 then FirstHVACIteration equals TRUE, if 0.0 then FirstHVACIteration equals false
+                        if (FirstHVACIteration) Par(2) = 1.0;
+                        Par(3) = ZoneNum;
+                        Par(4) = QZnReq;
+                        Par(5) = double(FanCoil(FanCoilNum).HeatCoilFluidInletNode);
+                        TempSolveRoot::SolveRoot(state, 0.001, MaxIterCycl, SolFlag, eHeatCoilCyclingR, CalcFanCoilHeatCoilPLRResidual, 0.0, 1.0, Par);
+                        if (SolFlag == -1) {
+                            ++FanCoil(FanCoilNum).ConvgErrCountH;
+                            if (FanCoil(FanCoilNum).ConvgErrCountH < 2) {
+                                ShowWarningError("Electric heating coil control failed in fan coil unit " + FanCoil(FanCoilNum).Name);
+                                ShowContinueError("  Iteration limit exceeded in calculating electric heating coil capacity modulation ");
+                                Calc4PipeFanCoil(state, FanCoilNum, ZoneNum, FirstHVACIteration, QUnitOut, _, eHeatCoilCyclingR);
+                                ShowContinueErrorTimeStamp("Load Request = " + TrimSigDigits(QZnReq) +
+                                                           ", Final Capacity = " + TrimSigDigits(QUnitOut));
+                                ShowContinueErrorTimeStamp("Electric heating coil part load ratio used during last iterations = " +
+                                                           TrimSigDigits(eHeatCoilCyclingR));
+                            } else {
+                                ShowRecurringWarningErrorAtEnd("Electric heating coil Iteration limit exceeded in fan coil unit " +
+                                                                   FanCoil(FanCoilNum).Name,
+                                                               FanCoil(FanCoilNum).MaxIterIndexH);
+                            }
+                        }
+                    } else {
+                        eHeatCoilCyclingR = 1.0;
+                    }
+                    // at the end calculate output
+                    if (FanCoil(FanCoilNum).FanOpMode == ContFanCycCoil) {
+                        Calc4PipeFanCoil(state, FanCoilNum, ZoneNum, FirstHVACIteration, QUnitOut, _, eHeatCoilCyclingR);
+                    } else {
+                        Calc4PipeFanCoil(state, FanCoilNum, ZoneNum, FirstHVACIteration, QUnitOut, PLR);
```

Comment:
This new code section handles the electric heating coil calculation. The *ehatiCoilCyclingR* modulates the heating coil capacity to meet the load when the fan operating mode is continuous and the coil is cycling.